tests/foreman/cli/test_template.py

```diff
@@ -74,6 +80,60 @@ def test_update_template_1(self):
         self.assertEqual(len(result.stderr), 0)
         self.assertEqual(updated_name, result.stdout['name'])
 
+    def test_create_template_with_location(self):
+        """@Test: Check if Template with Location can be created
+
+        @Feature: Template - Create
+
+        @Assert: Template is created and new Location has been assigned
+
+        """
+        try:
+            new_loc = make_location()
+            new_template = make_template({
+                'name': gen_string('alpha', 10),
+                'location-ids': new_loc['id'],
+            })
+        except CLIFactoryError as err:
+            self.fail(err)
+
+        self.assertIn(new_loc['name'], new_template['locations'])
+
+    def test_create_template_locked(self):
+        """@Test: Check that locked Template cannot be created
+
+        @Feature: Template - Create
+
+        @Assert: It is not allowed to create locked Template
+
+        """
+        with self.assertRaises(CLIFactoryError) as ex:
+            make_template({
+                'name': gen_string('alpha', 10),
+                'locked': 'true',
+            })
+        self.assertIn("This template is locked. Please clone it",
+                      ex.exception.message)
```

Comment:
I'm almost sure they will.

We are using factory, but maybe in this case should be used the "raw" way like:

```python
result = Template.create({'name':'somename', 'locked': True)
self.assertEqual(result.return_code, expected_returncode)
self.assertGreater(len(result.stderr), 0)
```

Also by using the assertRaises and with statement the lines 115 and 116 should be indented. Other similar tests just check if the exception is raised, but does not check the message or return code. With that said I think we can bring CLI tests to a next level again and check for its return code and if the command printed something, unfortunately we don't have a way to deal with the system being localized right now.